daemon/response.go

```diff
@@ -100,12 +94,8 @@ func (r *resp) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
 
 	hdr := w.Header()
 	if r.Status == http.StatusAccepted || r.Status == http.StatusCreated {
-		if m, ok := r.Result.(map[string]interface{}); ok {
-			if location, ok := m["resource"]; ok {
-				if location, ok := location.(string); ok && location != "" {
-					hdr.Set("Location", location)
-				}
-			}
+		if r.Meta.Change != "" {
+			hdr.Set("Location", changeURL(r.Meta.Change))
```

Comment:
I'm not sure what you mean by "a general response code path". It's the *only* response code path :-)

Setting the `Location` header is the recommended way to satisfy the

> response SHOULD include an indication of the request's current status and either a pointer to a status monitor or some estimate of when the user can expect the request to be fulfilled.

bit of the RFC for 202 (Accepted) responses, and 202 is the recommended response status for async operations that could still error.

We were lacking tests that ensured that we set this header; the code set it from the `"resource"` key which was set for all async ops before it was refactored into Meta. This change adds those tests, and changes the code to get it from Meta instead of resource.

Without the Location header we're less discoverable: somebody implementing a client of the rest api will have to know that all changes are at /v2/changes/{id} instead of getting it from the request (a fact we didn't document until Friday last week).